File: dongmin/week6/추억 점수.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];

        for (int i = 0; i < name.length; i++) {
            for (int j = 0; j < photo.length; j++) {
                List<String> strList = new ArrayList<>(Arrays.asList(photo[j]));    //photo에 크기가 전부 다르기에 ArrayList를 사용해 동적으로 크기 조절
                                                                                    //asList 메서드는 ArrayList에 현재 배열에 값을 넣어준다. 
                if (strList.contains(name[i])) {        //같은 값이 있을 경우
                    answer[j] += yearning[i];           //answer에 j번째 인덱스에 값을 yearning만큼 더함
                }
                strList.clear();                        //ArrayList를 초기화함
            }
        }

        return answer;
    }
}

File: yongkeun/6week/추억점수.js
function solution(name, yearning, photo) {
  var answer = [];
  for (let i = 0; i < photo.length; i++) {
    // i는 photo를 반복
    missingPoint = 0; // missingPoint를 0만큼 초기화
    for (let j = 0; j < name.length; j++) {
      // j는 name을 반복
      if (photo[i].includes(name[j])) {
        // 만약 photo 배열에 name 값이 있으면
        missingPoint += yearning[j]; // 현재 j index의 yearning을 plus
      }
    }
    answer.push(missingPoint); // missingPoint 를 answer에 push
  }
  return answer;
}
